msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/entityproviders/ForumsEntityProviderImpl.java

```diff
@@ -356,12 +356,12 @@ private Object getMessage(Long messageId,String siteId,String userId) {
 		
 	}
 	
-	private boolean checkAccess(BaseForum baseForum, String userId) {
+	private boolean checkAccess(BaseForum baseForum, String userId, String siteId) {
 		
 		if(baseForum instanceof OpenForum) {
 			
-			// If the supplied user is the super user, return true.
-			if(securityService.isSuperUser(userId)) {
+			// If the supplied user is the super user or an instructor, return true.
+			if(securityService.isSuperUser(userId) || forumManager.isInstructor(userId, "/site/" + siteId)) {
```

Comment:
I'd suggest creating a constant here out of: Entity.SEPARATOR + SiteService.SITE_SUBTYPE + Entity.SEPARATOR